src/main/java/io/vlingo/wire/fdx/bidirectional/BasicClientRequestResponseChannel.java

```diff
@@ -140,7 +139,9 @@ private void readConsume(final SocketChannel channel) throws IOException {
         pooledBuffer.release();
       }
     } catch (Exception e) {
-      pooledBuffer.release();
+      if (pooledBuffer.isInUse()){
```

Comment:
Is it safe to relesse() even if not in use? The release() should protect against releasing unused buffer (I think).

Reply:
That verification is there exactly because the `release()` throws an exception if the buffer is not in use:
```
    public void release() {
      if (!inUse.get()) {
        throw new IllegalStateException("Attempt to release unclaimed buffer: " + this);
      }
      notInUse();
    }
```

Reply:
Oh probably protecting against multiple release, but then it could actually release the wrong instance already in use by another thread. I don't want to go down this path now but maybe each actor must supply a unique id to access, use and release the buffer.